### DISCUSSION TITLE
Control Net not working through API?

### DISCUSSION THREAD

**Discussion by dbokser** [vladmandic/sdnext#640] [Q&A] [answered]
Hey all,
I'm developing a 3D software tool that's using the Automatic1111 API and I'd like to add support for this fork. Off the bat the API is connecting to vladamatic from the url, and it's working to generate images normally but control net isn't working at all. Here is the payload I'm sending along with my depth image that I'm inputting for the input_image. Anyone know what I'm doing wrong?

`{"prompt": "golden sphere, intricate detail", 
"negative_prompt": "", 
"steps": 10,
"seed": -1,
"width": 512,
"height": 512,
"sampler_index": "DPM++ 2M Karras",
"batch_size": 1,
"n_iter": 1,
"cfg_scale": 7,
"alwayson_scripts": 
	{"controlnet": 
		{"args": [{
			"lowvram": false,
			"module": "none",
			"model": "control_depth-fp16 [400750f6]",
			"weight": 1.0,
			"resize_mode": "Scale to Fit (Inner Fit)",
			"input_image": <input_image_string>,
			"processor_res": 512,
			"threshold_a": 100,
			"threshold_b": 255,
			"guidance_start": 0.0,
			"guidance_end": 1.0,
			"guessmode": false
			}]
		}
	}
}
`
![DepthPass_tmp](https://user-images.githubusercontent.com/2582947/235335422-95095b51-02ea-422b-aecc-dd14dc96efbd.png)

**Comment by finefin** [comment]
make sure 'Allow other script to control this extension' is enabled in ControlNet settings.

**Reply by dbokser** [reply]
it was unchecked, but didn't resolve the issue when I checked it.
![image](https://user-images.githubusercontent.com/2582947/235410175-8186d119-4d74-4382-8a42-ec7d8dd6db15.png)

**Comment by vladmandic** [comment, marked as answer]
There was an issue that I've resolved earlier today, can you update and try again?

**Reply by dbokser** [reply]
Yep, that did it. Works great now. Thanks!

**Comment by alloystorm** [comment]
Sorry if I'm interrupting, do you support the "--listen --api" switches that a1111 has?

We developed an Android frontend for SD A1111 that relys on the API access from local network to work. It would be good to support your fork as well.

**Reply by vladmandic** [reply]
api here is always-on so there is no switch. listen switch is supported.
